### PR TITLE
sumologicexporter: add graphite format

### PR DESCRIPTION
**Pull request by sumo-drosiek**
**Description:**
 - Add support for graphite format to sumo logic exporter
 - Add sanitization for carbon2 format
 - Replace `=` with `:` for sanitization

**Link to tracking Issue:**
#1498 

**Testing:**
Unit tests, manual tests

**Documentation:**
Code comments

### CODE REVIEW

**Comment by sumo-drosiek** [comment]
cc: @pmalek-sumo @pmm-sumo

**Comment by tigrannajaryan** [comment]
@pmm-sumo please review.

File: exporter/sumologicexporter/README.md [hunk 1]
```diff
@@ -10,11 +10,16 @@ Empty string means no compression
...
 - `metadata_attributes` (optional): List of regexes for attributes which should be send as metadata
 - `log_format` (optional) (logs only): Format to use when sending logs to Sumo. (default `json`) (possible values: `json`, `text`)
-- `metric_format` (optional) (metrics only): Format of the metrics to be sent (default is `prometheus`) (possible values: `carbon2`, `prometheus`)
-  `carbon2` and `graphite` are going to be supported soon.
+- `metric_format` (optional) (metrics only): Format of the metrics to be sent (default is `prometheus`) (possible values: `carbon2`, `graphite`, `prometheus`).
+- `graphite_template` (default=`%{_metric_}`) (optional) (metrics only): Template for Graphite format.
+Placeholders `%{attr_name}` will be replaced with attribute value for `attr_name`. `%{_metric_}` is going to be replaced with metric name.
+Applied only if `metric_format` is set to `graphite`.
 - `source_category` (optional): Desired source category. Useful if you want to override the source category configured for the source.
+Placeholders `%{attr_name}` will be replaced with attribute value for `attr_name`.
```
Comment (pmm-sumo) [inline exporter/sumologicexporter/README.md:18]: I think this is not entirely clear. Perhaps it should be marked as `graphite-only` or so?

Comment (sumo-drosiek) [reply exporter/sumologicexporter/README.md:18]: you mean `graphite_template`?

Comment (sumo-drosiek) [reply exporter/sumologicexporter/README.md:18]: `source_category` is independent of metric type

**Review by pmm-sumo — CHANGES_REQUESTED** [review]
I think it would be good to rephrase how attribute names are being expanded in `README.md`

File: exporter/sumologicexporter/README.md (hunk 1 above)
Comment (pmm-sumo) [reply exporter/sumologicexporter/README.md:18]: Oh, right, then I would rephrase the whole section that refers to it so it's clear what's going on. Perhaps something akin to:
> You can specify a template with an attribute for `source_category`, `source_name`, `source_host` or `graphite_template` using `%{attr_name}`. For example, `%{my_attr}`, when there is an attribute `my_attr: my_value` would be expanded to `my_value`. 

And then refer to it for each attribute, like:
> An attribute can be referred using `%{attr_name}`

**Comment by bogdandrutu** [comment]
@pmm-sumo please approve when you feel good about this PR so I can take a final look and merge

**Review by pmm-sumo — APPROVED** [review]

**Comment by pmm-sumo** [comment]
> @pmm-sumo please approve when you feel good about this PR so I can take a final look and merge

Thank you @bogdandrutu , I think we are good to merge now

**Review by tigrannajaryan — APPROVED** [review]
